src/compile/data/FilterInvalid.ts

```diff
@@ -0,0 +1,112 @@
+import {SCALE_CHANNELS} from '../../channel';
+import {ScaleType} from '../../scale';
+import {contains, Dict, differ, differArray, duplicate, extend, hash, keys, stringValue} from '../../util';
+import {VgFilterTransform, VgTransform} from '../../vega.schema';
+import {UnitModel} from './../unit';
+import {DataFlowNode} from './dataflow';
+import {isScaleChannel} from '../../channel'
+import {hasContinuousDomain} from '../../scale';
+import {QUANTITATIVE, TEMPORAL} from '../../type';;
+import {FieldDef} from '../../fielddef';
+import {Model, ModelWithField} from '../model';
+
+export class FilterInvalidNode extends DataFlowNode {
+  private _filter_null: Dict<FieldDef<String>>;
```

Comment:
1)  We don't use underscore scale naming.   
2) Don't add unnecessary `_` prefix for private if not necessary.
3) I think you can have one dict that stores {fieldDef, scaleType} and apply correct filter based on the scaleType.